### ISSUE TITLE
[Dropdown] aria-labelledby always set to a random value, breaking accessability tests

### ISSUE THREAD

**Issue by fburleigh** [zurb/foundation-sites#10763]
#### How to reproduce this bug:

    <a id="open-search-tool-bar" aria-label="Toggle search tool" data-toggle="search-tool">...</a>
    <div class="dropdown-pane" id="search-tool" data-dropdown></div>

#### What should happen:

`#search-tool`'s aria-labelledby attribute should be "open-search-tool-bar" from this phrase in dropdown.js:

    this.$currentAnchor.id || GetYoDigits(6, 'dd-anchor')

#### What happened instead:

After Foundation initializes, `#search-tool` has a an `aria-labelledby` attribute like 38f7-dd-anchor. Because that value does not match the data-toggle element, accessibility testing tools call the `div` element an error.

#### Analysis

`$currentAnchor` looks to be an array of elements (even if 1), so I think you want something like:

    this.$currentAnchor.get(0).attr('id') || GetYoDigits(6, 'dd-anchor')

I also think if I've already set aria-labelledby, you should leave it be.

#### Browser(s) and Device(s) tested on:

Firefox 58 and Safari 11.0.1.

#### Foundation Version(s) you are using:

6.4.3

**Comment by webuxr** [comment]
“I also think if I've already set aria-labelledby, you should leave it be.”

I’ll second that @fburleigh !

**Comment by ncoden** [comment]
This may be a duplicate of https://github.com/zurb/foundation-sites/issues/10627. So I'm closing this. Please reopen if you think this is not the same issue.

**Comment by webuxr** [comment]
@ncoden I just commented on that exact issue before seeing your response here. Thanks for consolidating.
